src/cargo/util/config.rs

```diff
@@ -31,11 +31,7 @@ pub struct Config {
 }
 
 impl Config {
-    pub fn new(shell: MultiShell) -> CargoResult<Config> {
-        let cwd = try!(env::current_dir().chain_error(|| {
```

Comment:
Its public, so it can and has to be called manually by any users of the cargo library that want to invoke the `ops::...` commands. In my specifc usecase I have one cargo instance as a library that I want to use to compile crates in different directories, for that I need to set the wd.

The way I see it, using the `env::cwd()` path as the wd for a operation is the choice of the invoker of the cargo operation, so cargo intern it's the responsibility of the routines that get called from the cargo binaries' main function.